### PR TITLE
Address suggested changes to input params

### PR DESCRIPTION
**Pull request by aeisenberg**
See #39

Biggest change is that we are now forcing more parameter families to 0. And we also need to account for nested arrays of parameters.

### CODE REVIEW

File: packages/mrc-ide-covidsim/src/imperial-params.ts [hunk 1]
```diff
@@ -274,6 +273,31 @@ export function assignAdminParameters(p: {}, subregionName: string) {
   p['Codes and country/province names for admin units'] = adminUnitNameLookup
 }
 
+// exported for testing
+export function setParameterFamilyTo0(
```
Comment (adityasharad) [inline packages/mrc-ide-covidsim/src/imperial-params.ts:277]:
```suggestion
/** Sets an entire family of parameters to zero.
 * Parameters are identified by substrings within the parameter name.
 * Values are set to zero or arrays of zeroes based on the expected shape.
 */
export function setParameterFamilyTo0(
```

File: packages/mrc-ide-covidsim/src/imperial-params.ts [hunk 2]
```diff
@@ -274,6 +273,31 @@ export function assignAdminParameters(p: {}, subregionName: string) {
...
+  periodCount: number
+) {
+  familyName = familyName.toLowerCase()
+  Object.entries(p)
+    .filter(([key]) => key.toLowerCase().includes(familyName))
+    .forEach(([key, value]) => {
+      if (Array.isArray(value)) {
+        // the top-level array should change size only if this is an _over time_ parameter
+        const newLen = /over time/i.test(key) ? periodCount : value.length
+        p[key] = new Array(newLen).fill(0)
```
Comment (adityasharad) [inline packages/mrc-ide-covidsim/src/imperial-params.ts:289]: Stylistic suggestion: pull out a function `zeroes(len: number): number[]` and use it here and below.

Comment (aeisenberg) [reply packages/mrc-ide-covidsim/src/imperial-params.ts]: sure

File: packages/mrc-ide-covidsim/src/imperial-params.ts (hunk 1 above)
Comment (aeisenberg) [reply packages/mrc-ide-covidsim/src/imperial-params.ts:282]: sure

**Pull request merged**